### DISCUSSION TITLE
dagmc-cubit meshing no longer working

### DISCUSSION THREAD

**Discussion by LukeLabrie** [svalinn/DAGMC#874] [Q&A]
I've used the dagmc-cubit toolchain before, primarily for meshing reactor geometries for simulation in openmc. I've recently come back to re-mesh some things, and I find that the toolchain is no longer working for me, but I am using the same version of cubit, and associated plugin. Has anything changed? Was hoping for some guidance in order to debug and hopefully get it working again. 

As an example, for the attached [step file](https://github.com/svalinn/DAGMC/files/11078385/step_file_and_jou_script.zip), the cubit-dagmc meshing step worked well for me in the past. Now, running the commands attached [here](https://github.com/svalinn/DAGMC/files/11078385/step_file_and_jou_script.zip) in the .jou output (also below), it gets completely stuck at 

```
Warning: Special case hit in merge_curves: circle
```

I even left it there for a day, and it was still stuck. Any idea what might be going on? Or debugging ideas?

Let me know if I can provide more detail. 

```
## Cubit
## Coreform Cubit Version 2021.11
## Coreform Cubit Build 9a3f8dc6e2
## Revised Wed Nov 17 19:27:08 2021 +0000
## Running 03/27/2023 12:53:48 PM
## Command Options:
import step "/home/luke/git/are/step_files/are_no_control_rods.step"
group "are_no_control_rods.step" add volume 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15
separate body all
validate volume all
volume 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 scale 101.54366
group "mat:inconel" add volume 1
group "mat:inconel" add volume 2
group "mat:inconel" add volume 3
group "mat:inconel" add volume 4
group "mat:insulation" add volume 5
group "mat:insulation" add volume 6
group "mat:salt" add volume 7
group "mat:BeO" add volume 8
group "mat:insulation" add volume 9
group "mat:insulation" add volume 10
group "mat:insulation" add volume 11
group "mat:coolant" add volume 12
group "mat:insulation" add volume 13
group "mat:stainless" add volume 14
group "mat:helium" add volume 15
create brick x 100000
create brick x 100005
subtract volume 16 from volume 17
group "mat:Graveyard" add volume 18
group "mat:helium_comp" add volume 18
imprint body all
healer autoheal body all
merge tolerance 0.0001
merge volume all group_results
set attribute on
```

**Comment by gonuke** [comment]
Sorry that this isn't working for you... I don't know what may have changed or why.  I'll ping @bohmt to see if he has any thoughts...

**Comment by bohmt** [comment]
I have loaded your step file and looked at your .jou file.  Some general advice is that **before** you imprint a model, you should have as "clean" and "defect free" a model as possible.  After loading your step file, if I run the cubit command:
_healer analyze vol all_
You will see cubit identifies many entity defects (84 bad Vertices, 106 bad Curves, 212 bad CoEdges, and these are in body: 1, 2, 14, 15)
It would be best to fully understand the nature of these defects but sometimes these defects can be repaired with "autoheal" .  For example:
_healer autoheal vol 1_
If you do this for each of the 4 bodies identified, then cubit identifies far fewer defects.
At this point, you should then imprint and merge the model.  You should (in theory) have a better merged model.
Note when merging, we typically use the command "merge all"
